include/nbl/builtin/hlsl/shapes/beziers.hlsl

```diff
@@ -13,11 +13,40 @@ namespace hlsl
 {
 namespace shapes
 {
+    // Modified from http://tog.acm.org/resources/GraphicsGems/gems/Roots3And4.c
+    // GH link https://github.com/erich666/GraphicsGems/blob/master/gems/Roots3And4.c
+    // Credits to Doublefresh for hinting there
+    // returns the roots, and number of filled in real values under numRealValues
+    float2 SolveQuadratic(float3 c, out int numRealValues)
+    {
+        // bhaskara: x = (-b ± √(b² – 4ac)) / (2a)
+        float b = c.y / (2 * c.z);
+        float q = c.x / c.z;
+        float delta = b * b - q;
+
+        if (delta == 0.0) // Δ = 0
+        {
+            numRealValues = 1;
+            return float2(-b, 0.0);
+        }
+        if (delta < 0) // Δ < 0 (no real values)
+        {
+            numRealValues = 0;
+            return 0.0;
+        }
+
+        // Δ > 0 (two distinct real values)
+        float sqrtD = sqrt(delta);
+        numRealValues = 2;
+        return float2(sqrtD - b, sqrtD + b);
+    }
+
     struct QuadraticBezier
```

Comment:
Make this 'QuadraticBezierOutline' with the sdf functions

Make another 'QuadraticBezier' with the evaluate and tangent etc functions and the ABCs 

QuadraticBezierOutline will contain a QuadraticBezier